build-support/bin/generate_github_workflows.py

```diff
@@ -293,14 +293,15 @@ def test_workflow_jobs(primary_python_version: str, *, cron: bool) -> Jobs:
             "runs-on": LINUX_VERSION,
             "needs": "bootstrap_pants_linux",
             "strategy": {"matrix": {"python-version": [primary_python_version]}},
+            "timeout-minutes": 30,
             "steps": [
                 *checkout(),
                 setup_toolchain_auth(),
                 *setup_primary_python(),
                 expose_all_pythons(),
                 pants_virtualenv_cache(),
                 native_engine_so_download(),
-                {"name": "Run Python tests", "run": "./pants test ::\n"},
+                {"name": "Run Python tests", "timeout-minutes": 40, "run": "./pants test ::\n"},
```

Comment:
It looks like you're doing 30 on the overall job, 40 on the step. Can remove this step timeout and do 40 for the overall job.